enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,806 @@
+   Users edits/creates the Route having Spec as follows -
+```yaml
+      apiVersion: route.openshift.io/v1
+      kind: Route
+      metadata:
+        labels:
+          type: sharded
+        name: hello-openshift-actions
+        namespace: hello-openshift
+      spec:
+        subdomain: hello-openshift
+        tls:
+          termination: edge
+        to:
+          kind: Service
+          name: hello-openshift
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+                set:
+                  value: DENY
+            - name: X-Cache-Info
+              action:
+                type: Set
+                set:
+                  value: "not cacheable; meta data too large"
+            - name: X-XSS-Protection
+              action:
+                type: Delete
+            - name: X-Source
+              action:
+                type: Set
+                set:
+                  value: "%[res.hdr(X-Value),lower]"
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name: X-SSL-Client-Cert
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der,base64]"
+            - name:   Content-Language
+              action:
+                type: Delete
+            - name: X-Target
+              action:
+                type: Set
+                set:
+                  value: "%[req.hdr(host),lower]"
+```
+
+#### Variation [optional]
+
+### API Extensions
+
+A) Set/Delete HTTP response headers in the frontend sections of haproxy.config via IngressController's Spec.
+The existing Ingress Controller API `HTTPHeaders *IngressControllerHTTPHeaders` is extended by adding a field for the header manipulations.
+
+```go
+// IngressControllerSpec is the specification of the desired behavior of the
+// IngressController.
+type IngressControllerSpec struct {
+    // httpHeaders defines policy for HTTP headers.
+    //
+    // If this field is empty, the default values are used.
+    //
+    // +optional
+    HTTPHeaders *IngressControllerHTTPHeaders `json:"httpHeaders,omitempty"`
+}
+```
+
+```go
+// IngressControllerHTTPHeaders specifies how the IngressController handles
+// certain HTTP headers.
+type IngressControllerHTTPHeaders struct {
+    // actions specifies options for performing action on headers.
+    // Note that this option only applies to cleartext HTTP connections
+    // and to secure HTTP connections for which the ingress controller
+    // terminates encryption (that is, edge-terminated or reencrypt
+    // connections).  Headers cannot be set or deleted for TLS passthrough
+    // connections.
+    // Setting HSTS header is supported to be set via route annotation haproxy.router.openshift.io/hsts_header and not via this API.
```

Comment:
I left a comment in the API.  Setting HSTS is a whole system in the IngressSpec.

Reply:
OK

Reply:
added here.